lib/logdna/client.rb

```diff
@@ -1,164 +1,132 @@
-
-    def check_side_buffer
-      return if @side_buffer.size == 0
-
-      @buffer.write(@side_buffer.string)
-      @side_buffer.truncate(0)
-      queued_side_messages = @side_messages
-      @side_messages = []
-      queued_side_messages.each { |message_hash_obj| @messages.push(message_hash_obj) }
-    end
-
-
-    # this should always be running synchronously within this thread
-    def buffer(msg, opts)
-      buffer_size = write_to_buffer(msg, opts)
-      unless buffer_size.nil?
-        process_buffer(buffer_size)
-      end
+    def schedule_flush
+      @flush_scheduled = true
+      start_timer = lambda {
+        sleep(@exception_flag ? @retry_timeout : @flush_interval)
+        flush if @flush_scheduled
+      }
+      thread = Thread.new { start_timer }
+      thread.join
     end
 
     def write_to_buffer(msg, opts)
-      return if msg.nil?
-      msg = encode_message(msg)
-
-      if @lock.locked?
-        @side_buffer.write(msg)
-        @side_messages.push(message_hash(msg, opts))
-        return
-      end
-
-      check_side_buffer
-      buffer_size = @buffer.write(msg)
-      @messages.push(message_hash(msg, opts))
-      buffer_size
-    end
-
-    def queue_to_buffer(queue=@queue)
-      next_object = queue.shift
-      write_to_buffer(next_object[:msg], next_object[:opts])
-    end
-
-    def process_buffer(buffer_size)
-      if buffer_size > @actual_byte_limit
-        @buffer_over_limit = true
-        flush()
-        @buffer_over_limit = false
+      if @lock.try_lock
+        processed_message = process_message(msg, opts)
+        new_message_size = processed_message.to_s.bytesize
+        @buffer.push(processed_message)
+        @buffer_byte_size += new_message_size
+        @lock.unlock
+
+        flush if @flush_limit <= @buffer_byte_size
+        schedule_flush unless @flush_scheduled
       else
-        @task = create_flush_task
+        @side_message_lock.synchronize do
+          @side_messages.push(process_message(msg, opts))
+        end
       end
     end
 
-    # this should be running synchronously if @buffer_over_limit i.e. called from self.buffer
-    # else asynchronously through @task
-    def flush()
-      if defined? @@request and !@@request.nil?
-        request_messages = []
-        @lock.synchronize do
-          request_messages = @messages
-          @buffer.truncate(0)
-          @messages = []
-        end
-        return if request_messages.empty?
+    def send_request
+      @side_message_lock.synchronize do
+        @buffer.concat(@side_messages)
+        @side_messages.clear
+      end
 
-        real = {
-          e: 'ls',
-          ls: request_messages,
-        }.to_json
+      @request.body = {
+        e: "ls",
+        ls: @buffer
+      }.to_json
 
-        @@request.body = real
-        @response = Net::HTTP.start(@uri.hostname, @uri.port, use_ssl: @uri.scheme == 'https') do |http|
-          http.request(@@request)
+      handle_excpetion = lambda do |message|
```

Comment:
Fix the spelling here and in every reference please

Reply:
done

Reply:
You said done... you didn't fix the spelling here and in every reference. Please do.

Reply:
I started to wonder whether I have dyslexia.